specification/cosmos-db/resource-manager/Microsoft.DocumentDB/2015-04-08/cosmos-db.json

```diff
@@ -437,26 +437,432 @@
     "/providers/Microsoft.DocumentDB/operations": {
       "get": {
         "tags": [
-            "Operations"
+          "Operations"
         ],
         "description": "Lists all of the available Cosmos DB Resource Provider operations.",
         "operationId": "Operations_List",
         "x-ms-examples": {
           "CosmosDBOperationsList": { "$ref": "./examples/CosmosDBOperationsList.json" }
         },
-        "parameters": [{
+        "parameters": [
+          {
             "$ref": "#/parameters/apiVersionParameter"
-        }],
+          }
+        ],
         "responses": {
-            "200": {
-                "description": "OK",
-                "schema": {
-                    "$ref": "#/definitions/OperationListResult"
-                }
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/OperationListResult"
             }
+          }
         },
         "x-ms-pageable": {
-            "nextLinkName": "nextLink"
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/{accountName}/metrics": {
+      "get": {
+        "operationId": "DatabaseAccounts_ListMetrics",
+        "x-ms-examples": {
+          "CosmosDBDatabaseAccountGetMetrics": { "$ref": "./examples/CosmosDBDatabaseAccountGetMetrics.json" }
+        },
+        "description": "Retrieves the metrics determined by the given filter for the given database account.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/accountNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/filterParameter"
+          }
+        ],
+        "responses": {
```

Comment:
looking at `MetricListResult`, it looks to me as if this operation should be marked as `x-ms-pageable` (with `nextLinkName` set to `null`)

Reply:
It's never going to return it as multiple pages

Reply:
that's why I suggested setting `nextLinkName` set to `null` 😉 it still improves generated code, since not a wrapper object with an array property is returned, but instead just the array itself. Of course, ignore my suggestion if you think there could be further properties appearing as siblings of `values` - since then, you would of course want users to see the "wrapper object"